### DISCUSSION TITLE
Point of Origin of an IFC element

### DISCUSSION THREAD

**Discussion by manas-v** [IfcOpenShell/IfcOpenShell#5019] [Q&A]
Hello!

I am creating a 3D model of a house using ifcopenshell, which includes walls, doors, and windows. Currently, the origin of the generated doors and windows is at the bottom corner of each element. I want to change the origin to the geometric center of these elements. How can I achieve this?

I am using "geometry.add_window_representation" to generate the window.

![image](https://github.com/user-attachments/assets/63c2e1c1-ebee-48f9-9147-766152c9c52c)

**Comment by aothms** [comment]
> Currently, the origin of the generated doors and windows is at the bottom corner of each element.

There are a lot of conventions around IFC geometry usage that enable robust exchanges and multi-disciplinary usage of the data. The API functions are created with these conventions in mind. I would advise to accept these as is. Some of these concepts are exemplified here https://standards.buildingsmart.org/documents/Implementation/IFC2x_Model_Implementation_Guide_V2-0b.pdf

**Reply by avinash-218** [reply]
> > Currently, the origin of the generated doors and windows is at the bottom corner of each element.
> 
> There are a lot of conventions around IFC geometry usage that enable robust exchanges and multi-disciplinary usage of the data. The API functions are created with these conventions in mind. I would advise to accept these as is. Some of these concepts are exemplified here https://standards.buildingsmart.org/documents/Implementation/IFC2x_Model_Implementation_Guide_V2-0b.pdf

Consider the scenario in which i create custom mesh using coordinates. In this scenario, is there any way to set the pivot point / origin of the custom mesh? For all custom mesh, the origin point is at 0,0,0 only. is there a way to change this point without translating the entire mesh itself?

**Comment by Moult** [comment]
It seems this was also posted here https://community.osarch.org/discussion/2263/point-of-origin-of-an-ifc-element#latest
